sale_blanket_order/models/blanket_orders.py

```diff
@@ -0,0 +1,649 @@
+    )
+
+    amount_untaxed = fields.Monetary(
+        string="Untaxed Amount",
+        store=True,
+        readonly=True,
+        compute="_compute_amount_all",
+        track_visibility="always",
+    )
+    amount_tax = fields.Monetary(
+        string="Taxes", store=True, readonly=True, compute="_compute_amount_all"
+    )
+    amount_total = fields.Monetary(
+        string="Total", store=True, readonly=True, compute="_compute_amount_all"
+    )
+
+    # Fields use to filter in tree view
+    original_uom_qty = fields.Float(
+        string="Original quantity",
+        compute="_compute_uom_qty",
+        search="_search_original_uom_qty",
+        default=0.0,
+    )
+    ordered_uom_qty = fields.Float(
+        string="Ordered quantity",
+        compute="_compute_uom_qty",
+        search="_search_ordered_uom_qty",
+        default=0.0,
+    )
+    invoiced_uom_qty = fields.Float(
+        string="Invoiced quantity",
+        compute="_compute_uom_qty",
+        search="_search_invoiced_uom_qty",
+        default=0.0,
+    )
+    remaining_uom_qty = fields.Float(
+        string="Remaining quantity",
+        compute="_compute_uom_qty",
+        search="_search_remaining_uom_qty",
+        default=0.0,
+    )
+    delivered_uom_qty = fields.Float(
+        string="Delivered quantity",
+        compute="_compute_uom_qty",
+        search="_search_delivered_uom_qty",
+        default=0.0,
+    )
+
+    def _get_sale_orders(self):
+        return self.mapped("line_ids.sale_lines.order_id")
+
+    @api.depends("line_ids")
+    def _compute_line_count(self):
+        self.line_count = len(self.mapped("line_ids"))
+
+    def _compute_sale_count(self):
+        for blanket_order in self:
+            blanket_order.sale_count = len(blanket_order._get_sale_orders())
+
+    @api.depends(
+        "line_ids.remaining_uom_qty", "validity_date", "confirmed",
+    )
+    def _compute_state(self):
+        today = fields.Date.today()
+        precision = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        for order in self:
+            if not order.confirmed:
+                order.state = "draft"
+            elif order.validity_date <= today:
+                order.state = "expired"
+            elif float_is_zero(
+                sum(order.line_ids.mapped("remaining_uom_qty")),
+                precision_digits=precision,
+            ):
+                order.state = "done"
+            else:
+                order.state = "open"
+
+    def _compute_uom_qty(self):
+        for bo in self:
+            bo.original_uom_qty = sum(bo.mapped("order_id.original_uom_qty"))
+            bo.ordered_uom_qty = sum(bo.mapped("order_id.ordered_uom_qty"))
+            bo.invoiced_uom_qty = sum(bo.mapped("order_id.invoiced_uom_qty"))
+            bo.delivered_uom_qty = sum(bo.mapped("order_id.delivered_uom_qty"))
+            bo.remaining_uom_qty = sum(bo.mapped("order_id.remaining_uom_qty"))
```

Comment:
Am I missing smth but this model doesn't have order_id field, so this method is doing nothing
seems you need extract sale.orders from  _get_sale_orders

Reply:
It might also be beneficial do do a single write/update instead of multiple assignments

```suggestion
            bo.update(
                {
                    "original_uom_qty": sum(bo.mapped("order_id.original_uom_qty")),
                    "ordered_uom_qty": sum(bo.mapped("order_id.ordered_uom_qty")),
                    "invoiced_uom_qty": sum(bo.mapped("order_id.invoiced_uom_qty")),
                    "delivered_uom_qty": sum(bo.mapped("order_id.delivered_uom_qty")),
                    "remaining_uom_qty": sum(bo.mapped("order_id.remaining_uom_qty")),
                }
            )
```